src/k_smollest.rs

```diff
@@ -0,0 +1,18 @@
+use std::collections::BinaryHeap;
+use std::cmp::Ord;
+
+pub(crate) fn k_smollest<T: Ord, I: Iterator<Item = T>>(mut iter: I, k: usize) -> BinaryHeap<T> {
```

Comment:
`smollest`

heh. I like it.

Reply:
I do not want to be a spoilsport, but I am wary with "`smollest`", as it e.g. disallows to `grep` for `smallest`, making it harder to find. Or is it a common term (e.g. know from other (programming) languages) familiar to most programmers?

Reply:
Hah, no this is definitely just cute and not a term of art. But alas, @phimuemue has a point.

@nbraud, I'm *so* sorry, but could you name this `k_smallest`?

Reply:
Done :)